### DISCUSSION TITLE
Kiali Dashboard Not able to fetch k8 Namespaces

### DISCUSSION THREAD

**Discussion by tppalani** [kiali/kiali#3931] [General]
I have successfully installed istio and deployed some sample app and application is up and running.
root@master:~# kubectl get pod
NAME                         READY   STATUS    RESTARTS   AGE
mydata-v1-847cd777c4-kc495   2/2     Running   0          39m
mydata-v2-65bbf55977-j67xp   2/2     Running   0          39m
myweb-66dc56ccd6-5g64b       2/2     Running   0          40m
Next i changed kiali dashboard cluster IP to Nodeport to access the dash brad from the browser

kubectl patch svc kiali -n istio-system --type='json' -p '[{"op":"replace","path":"/spec/type","value":"NodePort"},{"op":"replace","path":"/spec/ports/0/nodePort","value":30010}]'
Finally i can able to access the dashboard using node port with my host Ip http://machineip_port/ and could see my k8 namespaces without any apps please find the attached screen shot
![image](https://user-images.githubusercontent.com/6921037/115733853-dfd35600-a3a6-11eb-830b-0c85b3b24ff5.png)

Port 8080 and 15000 opned in open stack cloud network

**Comment by jshaughn** [comment]
Check the Kiali logs and also you may want to hover over that orange warning in the masthead (left of the message Bell icon).

**Comment by tppalani** [comment]
1) I kept my mouse point kiali dashboard overview option under this i could able to see the all the available namespaces from k8 cluster.

2) Note in all the namespace not able to see any application running list including istio-system and test namespace whereas i deployed my sample app init.

3) under the test namespace i select three dots to open my test namespace application but there is no running application.

![image](https://user-images.githubusercontent.com/6921037/115773476-fe991300-a3cd-11eb-9ecb-341e1b87b351.png)

test namespace empty

5) root@master:~# kubectl get all -n istio-system

NAME                                        READY   STATUS    RESTARTS   AGE

pod/grafana-784c89f4cf-cxpcz                1/1     Running   0          30h
pod/istio-egressgateway-bd477794-qv7n8      1/1     Running   0          31h
pod/istio-ingressgateway-79df7c789f-qlqcf   1/1     Running   0          31h
pod/istiod-6dc55bbdd-t5klg                  1/1     Running   0          31h
pod/jaeger-7f78b6fb65-xhz8j                 1/1     Running   0          30h
pod/kiali-dc84967d9-gw87t                   1/1     Running   28         30h
pod/prometheus-7bfddb8dbf-nd4gn             2/2     Running   35         30h

![image](https://user-images.githubusercontent.com/6921037/115773598-22f4ef80-a3ce-11eb-9407-219f877c700c.png)


Adding running container pod 2/2 with describe details one pod, i could two container one for nginx another is istio-proxy



Containers:

  nginx:

    Container ID:   docker://5b50075c22bebe7cfbe87ccab3168182eb75806933fbe99342ff43b101f7aade

    Image:          nginx

    Image ID:       docker-pullable://nginx@sha256:75a55d33ecc73c2a242450a9f1cc858499d468f077ea942867e662c247b5e412

    Port:           80/TCP

    Host Port:      0/TCP

    State:          Running

      Started:      Thu, 22 Apr 2021 10:38:08 +0000

    Ready:          True

    Restart Count:  0

    Environment:    <none>

    Mounts:

      /etc/nginx/conf.d/ from nginx-config (rw)

      /var/run/secrets/kubernetes.io/serviceaccount from default-token-bspz4 (ro)

  istio-proxy:

    Container ID:  docker://972e6e2a36bd877dae31aceb4e96eb9bd570dc71d4c99450201ef615ba97e07b

    Image:         docker.io/istio/proxyv2:1.9.1

    Image ID:      docker-pullable://istio/proxyv2@sha256:361ce2cb5b30094e9a08346aec2178a753f5844a3e7ed873c0deff2b4cae3607

    Port:          15090/TCP

    Host Port:     0/TCP

    Args:

      proxy

      sidecar

      --domain

      $(POD_NAMESPACE).svc.cluster.local

      --serviceCluster

      myweb.$(POD_NAMESPACE)

      --proxyLogLevel=warning

      --proxyComponentLogLevel=misc:error

      --log_output_level=default:info

      --concurrency

      2

    State:          Running

Adding Kiali Error log from the kiali pod in istio-system name space

-6dc55bbdd-t5klg: an error on the server ("unknown") has prevented the request from succeeding (get pods http:istiod-6dc55bbdd-t5klg:8080)

2021-04-22T19:18:22Z WRN GetPodProxyStatus is failing for [namespace: test] [pod: myweb-66dc56ccd6-5g64b]: Error fetching the proxy-status in the following pods: istiod-6dc55bbdd-t5klg: an error on the server ("unknown") has prevented the request from succeeding (get pods http:istiod-6dc55bbdd-t5klg:8080)

2021-04-22T19:18:22Z WRN GetPodProxyStatus is failing for [namespace: test] [pod: myweb-66dc56ccd6-5g64b]: Error fetching the proxy-status in the following pods: istiod-6dc55bbdd-t5klg: an error on the server ("unknown") has prevented the request from succeeding (get pods http:istiod-6dc55bbdd-t5klg:8080)

2021-04-22T19:18:22Z WRN GetPodProxyStatus is failing for [namespace: test] [pod: myweb-66dc56ccd6-5g64b]: Error fetching the proxy-status in the following pods: istiod-6dc55bbdd-t5klg: an error on the server ("unknown") has prevented the request from succeeding (get pods http:istiod-6dc55bbdd-t5klg:8080)

2021-04-22T19:18:22Z WRN GetPodProxyStatus is failing for [namespace: test] [pod: myweb-66dc56ccd6-5g64b]: Error fetching the proxy-status in the following pods: istiod-6dc55bbdd-t5klg: an error on the server ("unknown") has prevented the request from succeeding (get pods http:istiod-6dc55bbdd-t5klg:8080)

2021-04-22T19:18:22Z WRN GetPodProxyStatus is failing for [namespace: test] [pod: myweb-66dc56ccd6-5g64b]: Error fetching the proxy-status in the following pods: istiod-6dc55bbdd-t5klg: an error on the server ("unknown") has prevented the request from succeeding (get pods http:istiod-6dc55bbdd-t5klg:8080)

Adding kilai bell notification message

![image](https://user-images.githubusercontent.com/6921037/115773998-a282be80-a3ce-11eb-8cb5-76246935edb1.png)

**Comment by tppalani** [comment]
could you please help me on this etnire project team is waiting due to this issue.

**Comment by lucasponce** [comment]
Perhaps It is related to this ?

https://kiali.io/documentation/latest/faq/#google-cloud-private-cluster

**Reply by tppalani** [reply]
Hi required port 8080and 15000 opened already  https://kiali.io/documentation/latest/faq/#google-cloud-private-cluster what else I need to check

**Comment by jshaughn** [comment]
This is likely an environmental/configuration and not an issue with Kiali.  I will close this issue and convert to a discussion.

**Reply by tppalani** [reply]
See when I am removed side car injection lebel them I could see applications names from kiali dashboard,  then this is the not kiali issue.?

**Reply by lucasponce** [reply]
Hi @tppalani, I guess there is a combination of configuration steps that are affecting here.
Let me see if I can help you.

How did you install Istio and the AddOns in your cluster ?

I can try to see if I can reproduce the step where you are.

Also, which versions are you using ? I'm curious to see if with the kiali 1.33.1 you get the same errors.

**Reply by tppalani** [reply]
Hi @lucasponce  thank you so much for reaching out me.

1) version details

root@master1:/home/viswa# istioctl version
client version: 1.9.1
control plane version: 1.9.1
data plane version: 1.9.1 (5 proxies)

2) istio AddOns installed using below command where download the package using curl command

$ kubectl apply -f samples/addon

Thanks,
Palani

**Reply by lucasponce** [reply]
Thanks @tppalani.
To be honest, I have no clear picture about what is failing for you.
As my colleague @jshaughn has commented, it seems an infrastructure/configuration issue, Kiali can't connect to the sidecar of the pod and/or to connect to the istiod pod to check the status of the control plane.

If you're environment is a test one I'm going to share you a gist that I typically use to deploy a test cluster in a GCP platform:

https://gist.github.com/lucasponce/95f2d6ec8fd81eec32aec3102ef073e6

That configuration is tested, I'm sharing it with you with the only purpose to progress identifying what could be your problem.

(It is nothing special, just a summary of all steps described in istio.io / kiali.io)

That config proposes to expose Kiali through the Ingress instead to use directly nodePort. 
I don't think that is the cause, but I suspect that perhaps there were some steps in your cluster that probably it could affect.

Let me know if you can review it, play with it and get more new information.

If for a chance, you could reproduce the same problem in a platform that I could use, (i.e. minikube, kind, or GCP, OpenShift), I'd try to see if I can install it from my side following your steps and getting same problem.

Otherwise we need to chat to see if we can get any information that it helps us to identify what's going on.
